src/rotate.rs

```diff
@@ -19,14 +19,14 @@ impl<T> InvertableTransform for Rotate<T> {
     type Input = Vec<T>;
     type Output = Vec<T>;
 
-    fn forward(&self, mut value: Vec<T>) -> Vec<T> {
+    fn forward(&self, mut value: Vec<T>) -> Option<Vec<T>> {
         value.rotate_left(1);
-        value
+        Some(value)
```

Comment:
If there's no way for this to be a `None`, should it be an `Option`?

Reply:
It needs to be, because it is in the implementation of `InvertableTransform` which now expects this function to return an `Option`.